### DISCUSSION TITLE
Workspaces frustrating to work with

### DISCUSSION THREAD

**Discussion by RossSmyth** [ipetkov/crane#698] [General]
Hello,

I have Crane setup for a couple of my other single-crate projects and those have gone smoothly, but setting it up for my largest project that is workspace has been unpleasant. 

Mainly surrounding `pname`. I do not know what this does but as far as I can tell I can just set it to whatever and nothing changes. But even though it doesn't matter, I have been hitting errors about it all day. Even adding the `pname` field to the derivations doesn't always fix the issue. 

In my setup the only thing I care about are the executable and shared library outputs. To do this without nix, just cd in and run `cargo build`. It seems like it is trying to read the workspace toml to find some package name. But adding the fields lists in the manual doesn't fix anything.

I had it working at one point, but then when I followed the artifact reused manual page I now hit it again and haven't been able to fix it.

**Comment by RossSmyth** [comment]
Basically what is pname and how can I make crane shut-up about it 😄

**Comment by dpc** [comment]
Link to a project could help. It seems to me you're doing something fundamentally wrong, but it's hard to say what exactly.

**Reply by RossSmyth** [reply]
This is a proprietary project for my day job. I can post a sanitized version of the flake and describe the project.

**Comment by ipetkov** [comment]
`pname` is just the name of the "package" (or rather the *derivation*). It ultimately doesn't matter what it's called it's up to you, but it is required by nixpkgs's `mkDerivation`

Crane will try to guess a value for you (using the contents of Cargo.toml) if possible. Set it to  whatever non-null/non-empty value you want

**Reply by RossSmyth** [reply]
Sure, but setting the pname doesn't seem to do anything. I'll sanitize the flake and post it.

**Comment by RossSmyth** [comment]
Oh I was forgetting the `version` field as well. It could read that from `workspace.package.version` if it wanted but whatever.
